src/utils.rs

```diff
@@ -162,6 +167,45 @@ impl Array {
     }
 }
 
+/// The returned `mlx_vector_array` must be manually freed by the caller.
+pub(crate) unsafe fn new_mlx_vector_array(arrays: &[impl AsRef<Array>]) -> mlx_vector_array {
```

Comment:
Is it worth wrapping `mlx_vector_array` in an empty-ish Rust struct which implements Drop - just to save ourselves possibly forgetting to free? It seems so far, we use it in a method scope which would all end up dropping the created vector arrays.

Reply:
Let's do it then I'd say!

Reply:
Addressed in 55ca17d931a3dff70b189ecb3086957543ca4d55